3rdparty/mshadow/mshadow/base.h

```diff
@@ -1364,7 +1364,8 @@ struct minimum {
   default:                                          \
     LOG(FATAL) << "Unknown type enum " << type;     \
   }
-
+// amp_cast.h is using this MSHADOW_TYPE_SWITCH_WITH_BOOL in order to
+// avoid 'Unsupport enum type 12' error.
```

Comment:
no need to have this comment here as we don't leave comments on every usage of the macro

Reply:
Actually I put this comment as a TODO...
It is more harder than I thought to make bf16 work.
I had tried my best, but I have no idea about convolution ops.
This is a very very dirty fix and should be modified by another PR.

this PR only enable bf16 on dense layers
There are lots of works todo in the future.

Is it OK to mark a TODO on this comment?
Or simply delete this comment?

Reply:
I suggest to remove it and then track any todo as a separate issue.